src/sentry/monitors/validators.py

```diff
@@ -183,6 +183,17 @@ def validate_type(self, value):
             value = MONITOR_TYPES[value]
         return value
 
+    def validate_slug(self, value):
+        if not value:
+            return value
+
+        if Monitor.objects.filter(
+            slug=value, organization_id=self.context["organization"].id
+        ).exists():
+            raise ValidationError(f'The slug "{value}" is already in use.')
```

Comment:
This will help prevent most conflicts, but you could still have conflicts later as concurrent requests could both pass this check and then attempt to persist and fail.

Reply:
yeah we could slap a lock on this, but I'm not too worried about it

Reply:
Hopefully monitor creation isn't racy within an organization 🤞 if it is we'll find out :smile: